src/hyperlight_host/src/mem/shared_mem.rs

```diff
@@ -385,16 +408,65 @@ impl ExclusiveSharedMemory {
             return Err(MemoryRequestTooBig(total_size, isize::MAX as usize));
         }
 
-        // allocate the memory
+        let mut dwmaximumsizehigh = 0;
+        let mut dwmaximumsizelow = 0;
+
+        if std::mem::size_of::<usize>() == 8 {
+            dwmaximumsizehigh = (total_size >> 32) as u32;
+            dwmaximumsizelow = (total_size & 0xFFFFFFFF) as u32;
+        }
+
+        // Allocate the memory use CreateFileMapping instead of VirtualAlloc
+        // This allows us to map the memory into the surrogate process using MapViewOfFile2
+        let handle = unsafe {
+            CreateFileMappingA(
+                INVALID_HANDLE_VALUE,
+                None,
+                PAGE_EXECUTE_READWRITE,
```

Comment:
Does this need to be execute even when not using in-process mode?

Reply:
Good catch, fixed